uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs

```diff
@@ -63,236 +71,179 @@ pub struct KotlinWrapper<'a> {
 }
 impl<'a> KotlinWrapper<'a> {
     pub fn new(config: Config, ci: &'a ComponentInterface) -> Self {
+        // I _think_ I want to make this the holder of the LanguageOracle, which holds the CI.
+        // I haven't quite worked out if each type's CodeType should be one type (with defintion_code and lift/lower calling)
+        // or two (e.g. enums_::CodeType, enums_::CodeDeclType). a CodeDecl type would be an ideal place to put an askama Template definition,
```

Comment:
I also thought something along these lines when reading your dreamcode - the `CodeType` seemed to be doing double duty:

* The `declaration_code` method is intended to be called exactly once to render the code for implementing the actual type.
* The other methods are intended to be called multiple times at various other points in the rendering, to emit helper code that calls into parts of the declaration code.

Also, there are things for which we'll need declaration code which are not in any meaningful way a "type", such a function definitions. Splitting the declaration code from the type-helpers code seems like a promising approach to me.

Reply:
This approach is beginning to bare fruit.